tests/snuba/search/test_backend.py

```diff
@@ -2051,7 +2059,209 @@ def test_query(query):
             test_query(f"{key}:{val}")
 
 
-class CdcEventsSnubaSearchTest(TestCase, SnubaTestCase):
+class EventsTransactionsSnubaSearchTest(SharedSnubaTest):
+    @property
+    def backend(self):
+        return EventsDatasetSnubaSearchBackend()
+
+    def setUp(self):
+        super().setUp()
+        self.base_datetime = (datetime.utcnow() - timedelta(days=3)).replace(tzinfo=pytz.utc)
+
+        transaction_event_data = {
+            "level": "info",
+            "message": "ayoo",
+            "type": "transaction",
+            "culprit": "app/components/events/eventEntries in map",
+            "contexts": {"trace": {"trace_id": "b" * 32, "span_id": "c" * 16, "op": ""}},
+        }
+
+        def hack_pull_out_data(jobs, projects):
+            _pull_out_data(jobs, projects)
+            for job in jobs:
+                job["event"].groups = [self.group]
+            return jobs, projects
```

Comment:
I think that what we want here is to be able to use a different group depending on the transaction. So we could do this like

```
injected_group = None
def hack_pull_out_data(jobs, projects):
    _pull_out_data(jobs, projects)
    for job in jobs:
    job["event"].groups = [injected_group]
    return jobs, projects

...

injected_group =  self.perf_group_1 = self.create_group(type=GroupType.PERFORMANCE_SLOW_SPAN.value)
# Now when we call store_event the transaction will be created against `perf_group_1`.
        with mock.patch("sentry.event_manager._pull_out_data", hack_pull_out_data):
            transaction_event = self.store_event(
                data={
                    **transaction_event_data,
                    "event_id": "a" * 32,
                    "timestamp": iso_format(before_now(minutes=1)),
                    "start_timestamp": iso_format(before_now(minutes=1)),
                    "tags": {"my_tag": 1},
                },
                project_id=self.project.id,
            )

injected_group =  self.perf_group_2 = self.create_group(type=GroupType.PERFORMANCE_SLOW_SPAN.value)
...


```